struct_test.go

```diff
@@ -0,0 +1,27 @@
+package gofakeit
+
+import (
+	"fmt"
+	"testing"
+)
+
+type Foo struct {
+	s string
+	S string `fake:"{person.first}"`
+}
+
+type Sample struct {
+	A string  `fake:"{person.first}"`
+	P *string `fake:"{person.first}"`
+	s string
+	F *Foo `fake:""`
```

Comment:
Do we need to declare `fake:""`?

My thought would be that we fill the whole struct and only not fill a certain variable if the did something like `fake:omit` or whatever we think would be best.

Reply:
Explicit fake is necessary with at least string data type. Otherwise what random value do you fill it in with?
Keeping fake explicit is inline with json encoding and decoding built-in the default pkg.

Reply:
Actually I take that back. Let me give these a little bit more thought.
I like the `fake:"omit"` idea.

Reply:
👍

Reply:
@brianvoe updated, still need tests but I've included major support for all basic data types and pointers as well
String can be specify templates as well (using `Generate()`)

I'll start writing tests once we have this part squared off.